policy/approval/approve.go

```diff
@@ -128,15 +129,20 @@ func (r *Rule) IsApproved(ctx context.Context, prctx pull.Context) (bool, string
 		if err != nil {
 			return false, "", err
 		}
-
-		lastCommitTime := commits[len(commits)-1].CreatedAt
+		lastCommit := commits[len(commits)-1]
 
 		var allowedCandidates []*common.Candidate
 		for _, candidate := range candidates {
-			if candidate.CreatedAt.After(lastCommitTime) {
+			if candidate.CreatedAt.After(lastCommit.CreatedAt) {
 				allowedCandidates = append(allowedCandidates, candidate)
 			}
 		}
+
+		log.Debug().Msgf("discarded %d candidates invalidated by push of %s at %s",
+			len(candidates)-len(allowedCandidates),
+			lastCommit.SHA,
```

Comment:
Is this the full SHA of the commit or just substring of it? If it's the former, do we need to have the full SHA? May places like GitHub use the first 7-8 character

Reply:
This should be the full SHA: it comes from the API, which I believe only returns full values. Since this is for debugging, I don't mind having the full value, even if it makes the message longer.

Reply:
Cool, makes sense in this case